blueflood-core/src/main/java/com/rackspacecloud/blueflood/inputs/formats/JSONMetricsContainer.java

```diff
@@ -142,6 +147,13 @@ public void setTtlInSeconds(int ttlInSeconds) {
             this.ttlInSeconds = ttlInSeconds;
         }
 
+        public boolean isValid() {
+            long current = System.currentTimeMillis();
+
+            // valid if collectionTime is greater than past window or less than future window
+            return ((collectionTime >= current - pastDiff ) && ( collectionTime <= current + futureDiff ));
```

Comment:
As I look at the code more, I think having isValid() and getValidationErrors() will produce inconsistent results. Because each of those methods does its own call of System.currentTimeMillis(). 

Imagine us ingesting metrics that are on the borderline of being valid, just a few milliseconds shy of being invalid. When we call the isValid() on those, we get true. But a few milliseconds later, when we call getValidationErrors(), the System.currentTimeMillis() will be different, thus may cause those metrics to show up as being invalid. 

I think you should just keep one method to do the timestamp validation check. You will have to keep the invalid metrics as a class member variable, i.e: add:
```
List<Metric> invalidMetrics;
```
that gets populated when someone calls toMetrics(). And then you just do a getInvalidMetricsCount() or some method like that. If the count is greater than zero, return 207. If the count is all metrics are invalid, return 400. etc.